File: frontend/src/api/email.ts
import { api } from './client';

export interface EmailRequest {
  to: string;
  cc?: string;
  message?: string;
}

export interface EmailResponse {
  status: string; // "sent" | "queued" | "preview"
  subject: string;
  preview_html: string | null;
}

export async function sendWorkOrderEmail(
  workOrderId: string,
  data: EmailRequest,
): Promise<EmailResponse> {
  return api.post(`email/work-order/${workOrderId}`, { json: data }).json<EmailResponse>();
}

export async function sendInspectionEmail(
  inspectionId: string,
  data: EmailRequest,
): Promise<EmailResponse> {
  return api.post(`email/inspection/${inspectionId}`, { json: data }).json<EmailResponse>();
}

File: backend/app/schemas/email.py
from pydantic import BaseModel, EmailStr


class EmailRequest(BaseModel):
    to: EmailStr
    cc: EmailStr | None = None
    message: str | None = None  # Optional custom message from sender


class EmailResponse(BaseModel):
    status: str  # "sent" | "preview"
    subject: str
    preview_html: str | None = None  # Returned when SMTP not configured
